base/file.jl

```diff
@@ -81,37 +81,40 @@ Temporarily changes the current working directory and applies function `f` befor
 """
 cd(f::Function) = cd(f, homedir())
 
+checkmode(mode::Unsigned) = mode
+checkmode(mode::Signed) = throw(ArgumentError("mode must be an unsigned integer; try 0o$mode"))
```

Comment:
this seems unnecessarily strict. maybe just drop the error? (convert will check InexectError anyways)

Reply:
It also shouldn't suggest `0o` this way because `mode` won't be printed in octal.

Reply:
It probably makes sense to check that a mode only has three octal digits, but otherwise I think that any integer value between `0` and `511 = 0o777` should be acceptable, regardless of type.